packages/expo-gl/src/GLView.web.js

```diff
@@ -1,14 +1,198 @@
+
+  if (!gl._expo_texImage2D) {
+    gl._expo_texImage2D = gl.texImage2D;
+    gl.texImage2D = (...props) => {
+      let nextProps = [...props];
+      nextProps.push(_getImageForAsset(nextProps.pop()));
+      return gl._expo_texImage2D(...nextProps);
+    };
+  }
+
+  if (!gl._expo_texSubImage2D) {
+    gl._expo_texSubImage2D = gl.texSubImage2D;
+    gl.texSubImage2D = async (...props) => {
+      let nextProps = [...props];
+      nextProps.push(await _getImageForAsset(nextProps.pop()));
+      return gl._expo_texSubImage2D(...nextProps);
+    };
+  }
+
+  return gl;
+}
+export default class GLView extends React.Component<Props> {
+  static propTypes = {
+    onContextCreate: PropTypes.func,
+  };
+
+  state = {
+    width: undefined,
+    height: undefined,
+  };
+
+  nativeRef: ?HTMLCanvasElement;
+
+  static async createContextAsync() {
+    const { width, height, scale } = Dimensions.get('window');
+    const canvas = document.createElement('canvas');
+    canvas.width = width * scale;
+    canvas.height = height * scale;
+    return asExpoContext(canvas.getContext('webgl'));
+  }
+
+  static async destroyContextAsync(exgl: WebGLRenderingContext | ?number) {
+    const exglCtxId = getContextId(exgl);
+    if (exglCtxId in contextCache) {
+      document.removeChild(contextCache[exglCtxId]);
+      contextCache[exglCtxId] = undefined;
+    }
+  }
+
+  static async takeSnapshotAsync(
+    exgl: WebGLRenderingContext | ?number,
+    options: SnapshotOptions = {}
+  ) {
+    throw new UnavailabilityError('GLView', 'takeSnapshotAsync');
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._tryOnContextCreate();
+    window.addEventListener('resize', this._onLayout);
+  }
+  componentWillUnmount() {
+    window.removeEventListener('resize', this._onLayout);
+  }
+
+  _onLayout = () => {
+    this.setState({
+      width: this.width,
+      height: this.height,
+    });
+  };
+
+  get scale() {
+    return window.devicePixelRatio;
+  }
+
+  get x() {
+    return this.wrapperRef.screenX;
+  }
+  get y() {
+    return this.wrapperRef.screenY;
+  }
+
+  get width() {
+    if (!this.wrapperRef) return 1;
+    return this.wrapperRef.clientWidth;
+  }
+
+  get height() {
+    if (!this.wrapperRef) return 1;
+    return this.wrapperRef.clientHeight;
+  }
+
+  _tryOnContextCreate = () => {
+    if (this._isMounted && !!this.nativeRef && !!this.wrapperRef) {
+      this._onContextCreate();
+    }
+  };
+  _onContextCreate = async () => {
+    if (this._onContextCreated) return;
+
+    this._onContextCreated = true;
+    const { onContextCreate } = this.props;
+
+    if (onContextCreate) {
+      await new Promise(resolve => setTimeout(resolve));
+
+      const gl = asExpoContext(this.nativeRef.getContext('webgl2'));
+      onContextCreate(gl);
+    }
+  };
 
-export default class GLView extends React.Component<{}> {
   render() {
+    const { onContextCreate, sketch, style, ...props } = this.props;
```

Comment:
Is that because NCL is passing in props that shouldn't be passed in? If so, that's NCL's fault and GLView shouldn't be responsible for filtering that out.

Reply:
It's being passed in the processing library. I'll fix it there #3114